tools/highlighter-helper/src/client/styles/collapsed-lines.scss

```diff
@@ -0,0 +1,104 @@
+/* stylelint-disable scss/operator-no-newline-after */
+// collapsed lines
+div[class*='language-'].has-collapsed-lines {
+  &.collapsed {
+    overflow-y: hidden;
+    height: calc(
+      var(--vp-collapsed-lines) * var(--code-line-height) *
+        var(--code-font-size) + var(--code-padding-y) + 28px
+    );
+  }
+
+  .collapsed-lines {
+    --vp-collapsed-lines-bg: var(--code-c-bg);
+
+    position: absolute;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 4;
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    height: 28px;
+
+    background: linear-gradient(
+      to bottom,
+      transparent 0%,
+      var(--vp-collapsed-lines-bg) 55%,
+      var(--vp-collapsed-lines-bg) 100%
+    );
+
+    cursor: pointer;
+
+    transition: --vp-collapsed-lines-bg var(--vp-t-color);
+
+    &:hover {
+      --vp-collapsed-lines-bg: rgb(0 0 0 / 10%) !important;
+    }
+  }
+
+  &[data-highlighter='shiki'] .collapsed-lines {
+    --vp-collapsed-lines-bg: var(--code-c-bg, var(--shiki-light-bg));
+
+    [data-theme='dark'] & {
+      --vp-collapsed-lines-bg: var(--code-c-bg, var(--shiki-dark-bg));
+    }
+  }
+
+  .collapsed-lines::before {
+    --icon: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='1em' height='1em' viewBox='0 0 24 24'%3E%3Cpath fill='none' stroke='%23000' stroke-width='2' d='m18 12l-6 6l-6-6m12-6l-6 6l-6-6'/%3E%3C/svg%3E");
+    --vp-collapsed-lines-rotate: 0deg;
+
+    content: '';
+
+    display: inline-block;
+
+    width: 24px;
+    height: 24px;
+
+    background-color: var(--code-c-text);
+
+    mask-image: var(--icon);
+    mask-position: 50%;
+    mask-size: 20px;
+    mask-repeat: no-repeat;
+    pointer-events: none;
+
+    animation: code-collapsed-lines 1.2s infinite alternate-reverse ease-in-out;
+  }
+
+  &:not(.collapsed) {
+    code {
+      padding-bottom: max(var(--code-padding-y), 28px);
+    }
+
+    .collapsed-lines:hover {
+      --vp-collapsed-lines-bg: transparent !important;
+    }
+
+    .collapsed-lines::before {
+      --vp-collapsed-lines-rotate: 180deg;
+    }
+  }
+}
+
+@property --vp-collapsed-lines-bg {
+  inherits: false;
+  initial-value: #fff;
+  syntax: '<color>';
+}
```

Comment:
Is this needed?

https://developer.mozilla.org/zh-CN/docs/Web/CSS/@property From the table, firefox only supports it from 128. I am not sure if there will be errors under lower version of firfox.

Reply:
Don't worry, in lower versions it will just be recognized as a regular CSS variable, and after the downgrade, the corresponding background gradient transition animation simply won't take effect.